### PR TITLE
chapter/data: Restructure according to new methodology.

### PR DESCRIPTION
**Pull request by Grizzlly**
# Prerequisite Checklist

- [x] Read the [contribution guidelines](https://github.com/cs-pub-ro/operating-systems/blob/main/CONTRIBUTING.md#pull-requests) regarding submitting new changes to the project;
- [ ] Tested your changes against relevant architectures and platforms;
- [ ] Updated relevant documentation (if needed).

### CODE REVIEW

**Review by teodutu — CHANGES_REQUESTED** [review]
I made an initial pass highlighting the main things to do. Apply them to the other subchapter as well in a similar fashion. In addition, change your commit message following the requirements in the contributing guidelines [1]

[1] https://github.com/cs-pub-ro/operating-systems/blob/main/CONTRIBUTING.md#pull-requests

**Comment by teodutu** [comment]
In addition, there are a few false positives flagged by the Spellcheck action [1]. Add them to these wordlists [2], according to the topic to which they belong. After updating the wordlists, make sure you run `sort -u -o <wordlist_name.txt> <wordlist_name.txt>` to keep it sorted.

[1] https://github.com/cs-pub-ro/operating-systems/actions/runs/8376098698/job/22934978200?pr=71
[2] https://github.com/open-education-hub/actions/tree/main/spellcheck/config

**Comment by Grizzlly** [comment]
> I made an initial pass highlighting the main things to do. Apply them to the other subchapter as well in a similar fashion. In addition, change your commit message following the requirements in the contributing guidelines [1]
> 
> [1] https://github.com/cs-pub-ro/operating-systems/blob/main/CONTRIBUTING.md#pull-requests

Is `chapters/data` a good prefix? Or should it be more expressive like `chapters/data/drills`?

**Comment by Grizzlly** [comment]
@teodutu I have a problem when I build the site with the slides via `make`. This happens in the software stack too. I get a bunch of errors like this:

image

Any way to fix it?

**Comment by teodutu** [comment]
> > I made an initial pass highlighting the main things to do. Apply them to the other subchapter as well in a similar fashion. In addition, change your commit message following the requirements in the contributing guidelines [1]
> > [1] https://github.com/cs-pub-ro/operating-systems/blob/main/CONTRIBUTING.md#pull-requests
> 
> Is `chapters/data` a good prefix? Or should it be more expressive like `chapters/data/drills`?

`chapters/data` is good since you're going to add reading, slides, guides, media, drills etc.

**Comment by teodutu** [comment]
> @teodutu I have a problem when I build the site with the slides via `make`. This happens in the software stack too. I get a bunch of errors like this:
> 
> image
> 
> Any way to fix it?

The thing is that now because the paths to the media files have been moved relative to `slides.mdpp` (they used to be at the same level, but now they are one level lower - in the directories that correspond to each subchapter), `make html` doesn't work out of the box.

However, the builder used when you run `make build` in the root of the repo copies each subchapter's media files so the rendered website has correct paths. Therefore, you can use `make build` to view the slides.

File: chapters/data/investigate-memory/reading/investigate-memory.md [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+* memory access: read, write or execute
+* memory allocation
+* memory deallocation
+
+By far, the most important actions are allocation and deallocation.
+Because, if not done right, these can get to memory loss and poor memory use.
+
+Memory loss generally happens in the form of memory leaks.
+
+## Memory Leaks
```
[the remark this answers is not in the document]
Comment (Grizzlly) [reply chapters/data/investigate-memory/reading/investigate-memory.md:14]: In what sense?

File: chapters/data/investigate-memory/reading/investigate-memory.md [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+Enter the `support/memory-leak/` folder.
+It stores two files showing memory leaks:
+
+* one in C++: `memory_leak.cpp`
+* one in C: `memory_leak_malloc`
+
+Let's build and run the two executables:
+
+```console
+student@os:~/.../lab/support/memory-leak$ make
```
[the remark this answers is not in the document]
Comment (Grizzlly) [reply chapters/data/investigate-memory/reading/investigate-memory.md:29]: Shouldn't it be `drills/...` then?

File: chapters/data/arena/reading/arena.md
[the remark this answers is not in the document]
Comment (teodutu) [reply chapters/data/arena/reading/arena.md:1]: Yup. Each subsection in the Arena needs to be a separate task inside the `drills/tasks/` folder. It should be structured just like any other task:

```
task-name/
|-- README.md
|-- support/
`-- solution/
```
Keep the name of the tasks as it is in README.md, but call it `BONUS - Task Name` in `config.yaml` for now, so we know which was which.

File: chapters/data/investigate-memory/reading/investigate-memory.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (teodutu) [reply chapters/data/investigate-memory/reading/investigate-memory.md:14]: Move this section to `guides/memory-leaks/`. This folder should have the following structure:

```
memory-leaks/
|-- README.md
`-- support/
```

Guides are types of content that students are supposed to follow in a directed manner. This direction may come either from the teacher or the detailed descriptions in the `README.md` file. You can read about guides here: https://open-education-hub.github.io/methodology/Developing%20and%20Organizing%20Content/Guides/read

File: chapters/data/investigate-memory/reading/investigate-memory.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (teodutu) [reply chapters/data/investigate-memory/reading/investigate-memory.md:29]: Almost. It should be either: `tasks/...` or `guides/...`. I fixed my comment.

File: chapters/data/process-memory/reading/process-memory.md [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+```
+
+The resident size is `1968K`, much smaller than the virtual size.
+
+Note how each region has a size multiple of `4K`, this has to do with the memory granularity.
+The operating system allocates memory in chunks of a predefined size (in our case `4K`) called pages.
+
+[Quiz](../quiz/half-page.md)
+
+### Practice
```
Comment (teodutu) [inline chapters/data/process-memory/reading/process-memory.md:95]: Ditto: practice items should be made standalone under `drills/tasks/`.

**Review by teodutu — CHANGES_REQUESTED** [review]
I made another high-level pass. Try adding the new files to `config.yaml` to render them and see how things look on the website.

**Comment by teodutu** [comment]
Continued and implemented by #77.

**Pull request closed without merging**